pkg/util/utils.go

```diff
@@ -373,53 +384,48 @@ func UpdateK8sResourceAnnotation(client client.Client, ctx *context.Context, k8s
 	return nil
 }
 
-// GenerateID generate id for nsx resource, some resources has complex index, so set it type to string
-func GenerateID(res_id, prefix, suffix string, index string) string {
-	var id strings.Builder
-	if len(prefix) > 0 {
-		id.WriteString(prefix)
-		id.WriteString("_")
-	}
-
-	id.WriteString(res_id)
-	if len(index) > 0 {
-		id.WriteString("_")
-		id.WriteString(index)
+// GenerateIDByObject generate string id for nsx resource using the provided Object's name and uid
+func GenerateIDByObject(obj metav1.Object) string {
+	return normalizeNameByLimit(obj.GetName(), string(obj.GetUID()), common.MaxIdLength)
```

Comment:
so GenerateIDByObject is only used for cr, right? change it to GenerateCRIDByObject ?
GenerateID is used for other resource except CR?

Reply:
> so GenerateIDByObject is only used for cr, right?

Yes, this is my expectation. And this function is only used with VPC scenario. For T1, we still use GenerateID.
Another difference is, GenerateIDByObject has no prefix or suffix, it is only using CR_name-CR_uid. For some special cases which requiring prefix or suffix, we still use the GenerateID to wrap it.